server/jetstream_api.go

```diff
@@ -952,7 +952,14 @@ func (s *Server) jsStreamInfoRequest(sub *subscription, c *client, subject, repl
 		s.sendAPIResponse(c, subject, reply, string(msg), s.jsonResponse(&resp))
 		return
 	}
-	resp.StreamInfo = &StreamInfo{Created: mset.Created(), State: mset.State(), Config: mset.Config()}
+	config := mset.Config()
+	// MQTT streams are created without subject, but "nats" tooling would then
+	// fail to display them since it uses validation and expect the config's
+	// Subjects to not be empty.
```

Comment:
technically the nats utility is wrong, since the server accepts it and it then defaults to stream name. I could allow it but at the time decided it would be better to be explicit

Reply:
I will need, as part of clustering work, to have streams not visible to the outside and used for internal purposes as well for a WAL.